src/main/java/com/gentics/elasticsearch/client/okhttp/ElasticsearchOkClient.java

```diff
@@ -175,9 +175,10 @@ public Single<T> executeAsync(Request request) {
 			call.enqueue(new Callback() {
 				@Override
 				public void onFailure(Call call, IOException e) {
-					// Don't call the onError twice. Cancelling will trigger another error.
-					if (!"Canceled".equals(e.getMessage())) {
-						sub.onError(e);
+					// Don't call the onError twice. 
+					if (!sub.isDisposed()) {
```

Comment:
Here, in case of `Cancelled` was coming from elsewhere, we skip its processing, letting the handler thread hanging forever.

Reply:
I have experimented with this part, removing any error response. Got results, similar to the ones from the ticket - no errors, no logs, no calls. So I'm almost sure the ignorance of the `Canceled` error is the key.